src/ucp/core/ucp_ep.c

```diff
@@ -1143,6 +1143,29 @@ static void ucp_ep_set_lanes_failed(ucp_ep_h ep, uct_ep_h *uct_eps)
     }
 }
 
+static int ucp_can_uct_ep_unprogress(ucp_ep_h ep, uct_ep_h uct_ep)
+{
+    return ep->worker->context->config.ext.adaptive_progress &&
+           /* Do not unprogress an already failed lane */
+           !ucp_is_uct_ep_failed(uct_ep) &&
+           !ucp_wireup_ep_test(uct_ep);
+}
+
+void ucp_ep_unprogress_uct_ep(ucp_ep_h ep, uct_ep_h uct_ep,
+                              ucp_rsc_index_t rsc_index)
+{
+    ucp_worker_iface_t *wiface;
+
+    if (rsc_index == UCP_NULL_RESOURCE ||
+        !ucp_can_uct_ep_unprogress(ep, uct_ep)) {
```

Comment:
ucp_can_uct_ep_unprogress used only once - can remove the func and put its body here

Reply:
done